Modules/Devices/src/Devices.Infrastructure/PushNotifications/PushService.cs

```diff
@@ -49,12 +78,19 @@ private async Task SendNotification(IEnumerable<PnsRegistration> registrations,
 
             var pnsConnector = _pnsConnectorFactory.CreateFor(platform);
```

Comment:
Isn't it sufficient to have it resolved once per group key before the foreach loop?

Reply:
The loop is for iterating the groups. And each group has a different platform. So I don't see a way for optimization here. I might miss something though.